daemon/cluster/swarm.go

```diff
@@ -25,19 +25,15 @@ import (
 func (c *Cluster) Init(req types.InitRequest) (string, error) {
 	c.controlMutex.Lock()
 	defer c.controlMutex.Unlock()
-	c.mu.Lock()
 	if c.nr != nil {
 		if req.ForceNewCluster {
 			if err := c.nr.Stop(); err != nil {
```

Comment:
I know it is ugly but can we take a `c.mu.Lock()` temporary here so that we wait for API handlers to finish up.